materialize/resources/utils.go

```diff
@@ -62,14 +62,18 @@ func QuoteString(input string) (output string) {
 }
 
 func QuoteIdentifier(input string) (output string) {
-	output = `"` + strings.Replace(input, `"`, `""`, -1) + `"`
+	parts := strings.Split(input, ".")
+	for i, p := range parts {
+		parts[i] = `"` + strings.Replace(p, `"`, `""`, -1) + `"`
+	}
+	output = strings.Join(parts, ".")
```

Comment:
This indicates that something is getting conflated. Identifiers must _not_ split on `.`s, as you can say e.g.

```
CREATE TABLE "a.b.c" ...
```

which is quite different from:

```
CREATE TABLE "a"."b"."c" ...
```

Do you know what required this? Probably needs to be solved in another way.

Reply:
Ahh, I see, it's the issue I brought up [here](https://github.com/MaterializeInc/terraform-provider-materialize/issues/25#issuecomment-1431892531):

> One additional wrinkle here: referring to a secret in Terraform land is hard! The way the e.g. Snowflake provider handles object references is to take the database, schema, and name as three separate parameters. See, for example, the [`snowflake_table`](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/table_grant) resource. The way that would look here is (brace yourself):
> 
> ```terraform
> ssl_certificate = "literal string"
> # OR
> ssl_certificate_secret_database_name = "db"
> ssl_certificate_secret_schema_name = "schema"
> ssl_certificate_secret_name = "sekret"
> ```
> 
> I can't see a way to do it with a single `ssl_certificate_secret` field name that doesn't have a bunch of pitfalls around identifier quoting.

I think we need to sort this out, or we will forever have weird edge cases in what sort of names are allowed and what sort of quoting rules you have to follow. 

One way to make this more ergonomic is to use nested schemas, so you'd say:

```
resource "materialize_source_postgres" "example_source_postgres" {
  name                = "source_postgres"
  schema_name         = "schema"
  postgres_connection {
    # Optionally choose a different schema. (Defaults to the source's schema?)
    # schema_name = "somewhere_else"
    name = "pgconn"
  }
}
```

Reply:
Hm yes good point, in this this seems to be non-trivial, as if the user passes database.schema.table, then if we were to just quote it with `CREATE TABLE "database.schema.table"` that will result in a table named `database.schema.table` rather than a table in the `"database"."schema"` schema.

Will need to rethink this as at the moment if someone has a dot in their identifier name, it will mess up the logic. Open to suggestions.

Reply:
Snowflake have a similar approach:

https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/pkg/snowflake/escaping.go#L39-L52

But they have an extra `MustCompile` check inside that `for` loop

Reply:
I've just pushed that change for the `postgres_connection`, let me know what you think! 

Basically, I've abstracted the nested schema into a separate func so that it can be re-used among all identifiers.

Once we agree upon a concept, I will carry over the change to the rest of the identifiers.

Reply:
All identifiers have now been updated and use the new syntax outlined in @benesch example above! Let me know what you think.